action/protocol/context.go

```diff
@@ -105,7 +105,7 @@ type (
 		RevertLog                   bool
 		TolerateLegacyAddress       bool
 		ValidateRewardProtocol      bool
-		CreateZeroNonceAccount      bool
+		CreateLegacyNonceAccount    bool
```

Comment:
can we still use the `CreateZeroNonceAccount` flag and option? this is more natural (handling a new type, `flag = false` and no option is default), and we don't need to go over all the changes again

Reply:
and we can continue to use `state.NewEmptyAccount()`, that is a good API with clean name/intention

Reply:
no
1. a new and empty account is height dependent as well
2. default implementation should create what is expected by default. for this case, we expect the new account to be created with init pending nonce = 0. To create an account with nonce = 1, an option is needed.